### ISSUE TITLE
First impressions

### ISSUE THREAD

**Issue by Mattk70** [birdnet-team/birdnet-V3.0-dev#1]
Thank you for sharing this repo! I've cloned it and got things running on Windows 11.

You guys have done a great job! I already like the new streamlit web app more than the current BNA GUI.

**Performance (speed):** for longer files (I tried 1 & 3 hours) it's really quick, about 50% faster inference than I have BirdNET 2.4 running on TFJS. Spectrogram generation is _much_ quicker than BirdNET Analyzer GUI.

**Compatibility**: it runs out of the box, although on Windows the Python venv puts exe's in a Scripts folder so the README might want to say that, on windows, the command is:

`source .venv/Scripts/activate `

Other things I noticed...
1. M4A doesn't appear to be supported, contrary to the docs
2. Streamlit's 200MB limit is a bit low for wav files, can it be raised?
3. Precision is quite poor - but as you haven't added much by way of anthropogenic noise in the training data, that's to be expected for my environment.
4. Conversely, recall seems very good!
5. Strix aluco (Tawny Owl) is still a problem

I hacked the app.py to spit out Audacity labels for results in addition to CSV, and throw in a couple of timers. Having the output as audacity makes it a lot easier to compare results with other models, at least subjectively. Let me know if you want a PR?

### Other thoughts

You've switched to Torch! How do you find it? As for GPU inference, CUDA is a PITA to set up, but WebGPU is a breeze - are you planning a JS version? I can look at torch.js or js-pytorch if you haven't thought much about that yet.

**Comment by Mattk70** [comment]
> Streamlit's 200MB limit is a bit low for wav files, can it be raised?

I found out you can set the max file size by calling streamlit with the following option:

`streamlit run app.py --server.maxUploadSize 2048` <- this allows 2GB

**Comment by kahst** [comment]
Thanks for the feedback, Matt! 

- hadn't tried M4a, just assumed Librosa would be able to handle it, need to change the description then
- I just used the default Streamlit file size, but will certainly add a note on how to process larger files. I want to host the demo on one of our servers at some point (for experimentation not necessarily serious data processing) and bigger file sizes might create issues
- I definitively would want a TFJS version (along with many other formats); we'll try to provide a dedicated version (not sure if ONNX is helpful with that or not)
- I don't have any experience with Torch, Mario has been using it for quite a while and that's why he felt more comfortable compared to TF; but overall I believe TF is not the future (except for deployment because of TFLite and TFMicro and TFJS)
